service/controller/resource/tccp/create.go

```diff
@@ -632,10 +632,6 @@ func (r *Resource) newParamsMainSubnets(ctx context.Context, cr infrastructurev1
 
 	var privateSubnets []template.ParamsMainSubnetsSubnet
 	for _, az := range zones {
-		if az.Name != key.MasterAvailabilityZone(cr) {
-			continue
-		}
-
```

Comment:
we need to create all private subnets for all AZs not only for the first

Reply:
this make things much easier

Reply:
Why is this now though? Is this just for simplicity and we do not care or is there a hard requirement for that?

Reply:
This is a hard requirement, as the condition was cutting master subnets to only 1 private subnet  without any regard to what it really needs to be with 3 masters